### PR TITLE
feat: add `rocks list` command

### PR DESCRIPTION
**Pull request by vhyrro**
This PR sets up the necessary boilerplate to list out all available rocks. Listing outdated rocks requires more work and will be worked on separately.

### CODE REVIEW

File: rocks-bin/src/list.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            .as_ref()
+            .ok_or_eyre("lua version not supplied!")?,
+    )?;
+    let available_rocks = tree.list();
+
+    // TODO(vhyrro): Add `outdated` support.
+
+    if list_data.porcelain {
+        for (name, versions) in available_rocks {
+            println!("{}: {}", name, versions.join(" "));
```
Comment (mrcjkb) [inline rocks-bin/src/list.rs:29]: suggestion (non-blocking): What if we ditch the luarocks "porcelain" output in favour of json?

**Review by mrcjkb — APPROVED** [review]

File: rocks-bin/src/list.rs (hunk 1 above)
Comment (vhyrro) [reply rocks-bin/src/list.rs:29]: Love the sound of that idea. I'll keep the `--porcelain` name but change the internals to produce JSON instead.

File: rocks-bin/src/list.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+use clap::Args;
+use eyre::{OptionExt as _, Result};
+use itertools::Itertools as _;
+use rocks_lib::{config::Config, tree::Tree};
+use text_trees::{FormatCharacters, StringTreeNode, TreeFormatting};
+
+#[derive(Args)]
+pub struct List {
```
Comment (mrcjkb) [inline rocks-bin/src/list.rs:8]: suggestion: name this `ListCmd` to clarify that this isn't a list. 
Clap has macros that let you specify the name on the CLI

File: rocks-bin/src/list.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    #[arg(long)]
+    outdated: bool,
+    #[arg(long)]
+    porcelain: bool,
+}
+
+pub fn list(list_data: List, config: &Config) -> Result<()> {
```
Comment (mrcjkb) [inline rocks-bin/src/list.rs:15]: suggestion:

```suggestion
pub fn list_installed(list_data: List, config: &Config) -> Result<()> {
```

File: rocks-bin/src/list.rs (hunk 2 above)
Comment (vhyrro) [reply rocks-bin/src/list.rs:8]: Yeah good call, I'll change that.

**Pull request merged**